geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java

```diff
@@ -8,32 +8,57 @@
 
 package org.locationtech.geomesa.hbase.filters;
 
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import org.apache.hadoop.hbase.Cell;
 import org.apache.hadoop.hbase.exceptions.DeserializationException;
 import org.apache.hadoop.hbase.filter.FilterBase;
 import org.apache.hadoop.hbase.util.Bytes;
 import org.geotools.filter.text.cql2.CQLException;
 import org.geotools.filter.text.ecql.ECQL;
 import org.locationtech.geomesa.features.interop.SerializationOptions;
+import org.locationtech.geomesa.features.kryo.KryoBufferSimpleFeature;
 import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer;
 import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes;
-import org.opengis.feature.simple.SimpleFeature;
 import org.opengis.feature.simple.SimpleFeatureType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 
 public class JSimpleFeatureFilter extends FilterBase {
     private String sftString;
     private SimpleFeatureType sft;
-    private KryoFeatureSerializer serializer;
 
     private org.opengis.filter.Filter filter;
     private String filterString;
-
-    public JSimpleFeatureFilter(String sftString, String filterString) {
+    private KryoBufferSimpleFeature reusable;
+
+    private static Logger log = LoggerFactory.getLogger(JSimpleFeatureFilter.class);
+
+    private static LoadingCache<String, SimpleFeatureType> sfts =
```

Comment:
Could we factor out org.locationtech.geomesa.accumulo.iterators.IteratorCache to geomesa-utils and re-use it here?